### PR TITLE
Avoid enqueuing tree mutation record when there is no mutation

### LINKED ISSUE #796: Avoid enqueuing tree mutation record when there is no mutation
Currently, the [replace all](https://dom.spec.whatwg.org/#concept-node-replace-all) steps queues a tree mutation record even when `addedNodes` and `removedNodes` are empty. When following the spec strictly, the [childList Node.textContent: textContent no mutation](https://github.com/web-platform-tests/wpt/blob/master/dom/nodes/MutationObserver-childList.html#L103-L107) fails.

IMHO, the replace all step should only enqueue a tree mutation record when either `addedNodes` or `removedNodes` is not empty. In fact, servo implementation already add [this extra branch check](https://github.com/servo/servo/blob/5ab2d4ea835374c2ca88f5d9b50100669f142273/components/script/dom/node.rs#L2008-L2016).

### PR DESCRIPTION
**Pull request by shvaikalesh**
Fixes #796.

***
<a href="https://whatpr.org/dom/822.html" title="Last updated on Jan 22, 2020, 3:23 PM UTC (de10bc3)">Preview</a> | <a href="https://whatpr.org/dom/822/55379a7...de10bc3.html" title="Last updated on Jan 22, 2020, 3:23 PM UTC (de10bc3)">Diff</a>

### CODE REVIEW

File: dom.bs [hunk 1]
```diff
@@ -2604,7 +2604,8 @@ within a <var>parent</var>, run these steps:
  <li><p>If <var>node</var> is non-null, then <a for=/>insert</a> <var>node</var> into
  <var>parent</var> before null with the <i>suppress observers flag</i> set.
 
- <li><p><a>Queue a tree mutation record</a> for <var>parent</var> with <var>addedNodes</var>,
+ <li><p>If either <var>addedNodes</var> or <var>removedNodes</var> <a for=list>is not empty</a>,
```
Comment (annevk) [inline dom.bs]:
```suggestion
 <li><p>If either <var>addedNodes</var> or <var>removedNodes</var> <a for=set>is not empty</a>,
```

File: dom.bs [hunk 2]
```diff
@@ -3591,6 +3592,9 @@ run these steps:
 <var>target</var> with null, null, null, <var>addedNodes</var>, <var>removedNodes</var>,
 <var>previousSibling</var>, and <var>nextSibling</var>.
 
+<p class="note no-backref">Either <var>addedNodes</var> or <var>removedNodes</var>
+<a for=list>is not empty</a>.
```
Comment (annevk) [inline dom.bs]: This seems nice to call out, but then we should turn the thing above this into an algorithm with two steps and the first step is this, as an assert.

File: dom.bs (hunk 1 above)
Comment (shvaikalesh) [reply dom.bs]: Please also see #823.

File: dom.bs (hunk 2 above)
Comment (shvaikalesh) [reply dom.bs]: Resolved in de10bc3.

**Comment by annevk** [comment]
Thanks @shvaikalesh, really appreciate it!

**Pull request merged**